### PR TITLE
Use injected ExecutorService instead of creating a new one

### PR DESCRIPTION
**Pull request by cberner**

### CODE REVIEW

**Comment by martint** [comment]
Explain why in the commit message

File: presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java [hunk 1]
```diff
@@ -453,7 +453,8 @@ protected final void setup(String host, int port, String databaseName, String ti
                 true,
                 typeManager,
                 locationService,
-                partitionUpdateCodec);
+                partitionUpdateCodec,
+                newCachedThreadPool());
```
Comment (dain) [inline presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java]: We should put a "reasonable" limit on this.

**Comment by dain** [comment]
Other than my one comment and martin's comment, looks good

File: presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java (hunk 1 above)
Comment (electrum) [inline presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java]: Does this matter for a test?

**Comment by cberner** [comment]
Since the leak is only in our internal connector, I'm not sure what to say in this commit message. Something like "This allows connectors to provide their own ExecutorService"?

**Comment by martint** [comment]
What causes the leak? Maybe something like "not having this forces connectors that extend this one to do xxxx which results in leaks."

**Comment by cberner** [comment]
It's the way our internal client `Metadata` is written. We create a new `HiveMetadata` object for every metadata call, which in turn creates a new `ThreadFactory` with its own `ThreadGroup`. I'll say "Not having this forces connectors that extend this one to leak ThreadGroups, if they don't reuse HiveMetadata objects."

**Pull request merged**